### ISSUE TITLE
Add explicit documentation regarding the ordering of Union types for serialization/deserialization

### ISSUE THREAD

**Issue by john-bodley** [lovasoa/marshmallow_dataclass#183]
I was wondering whether there was merit in adding more documentation regarding the `Union` type, specifically related to the order of types is relevant—this is somewhat counter intuitive as the term _union_ applies to set theory where sets are unordered. 

For example say you have a field which could be a boolean or float, then the following snippet,

```python
from dataclasses import dataclass
from typing import Union

import marshmallow_dataclass


@dataclass
class Foo:
    bar: Union[bool, float]


foo_schema = marshmallow_dataclass.class_schema(Foo)()
foo = foo_schema.load({"bar": 1.0}) 
```

returns `Foo(bar=True)` rather than `Foo(bar=1.0)`. The fix is merely to redefine the `bar` field as `Union[float, bool]` given that the deserialization attempt is based on the order of the defined unioned types.

**Comment by lovasoa** [comment]
Yes, you are right, the documentation is not very clear about how Unions work. PR welcome !
